libraries/default.rb

```diff
@@ -0,0 +1,68 @@
+def collect_github_repositories(github_token, org)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
+
+  require 'octokit'
+
+  github = Octokit::Client.new(access_token: github_token)
+  github.auto_paginate = true
+
+  github.org_repos(org).map(&:name)
+end
+
+def public_address
+  node.fetch('cloud', {}).fetch('public_ipv4', nil) || node['fqdn']
+end
+
+# rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+def set_up_github_push(github_token, org_name, repo_name, job_name, trigger_token)
+  if ::ObjectSpace.const_defined?('Chef')
+    # Chef exists! Lets make sure Octokit is installed.
+    chef_gem 'octokit'
+  end
```

Comment:
This gem is already being installed before this function is being called so we should remove this redundancy.